test-data/unit/check-parameter-specification.test

```diff
@@ -1520,3 +1520,28 @@ def identity(func: Callable[P, None]) -> Callable[P, None]: ...
 @identity
 def f(f: Callable[P, None], *args: P.args, **kwargs: P.kwargs) -> None: ...
 [builtins fixtures/paramspec.pyi]
+
+[case testParamSpecificationSurvivesCall]
+# https://github.com/python/mypy/issues/12595
+from typing import TypeVar, Callable
+from typing_extensions import Concatenate, ParamSpec
+
+T = TypeVar('T')
+U = TypeVar('U')
+P = ParamSpec('P')
+
+def t(f: Callable[P, T]) -> Callable[P, T]: ...
+
+def pop_off(
+    transform: Callable[Concatenate[T, P], U]
+) -> Callable[P, Callable[[T], U]]: ...
+
+u = pop_off(t)
```

Comment:
TBH I don't think this situation is what is going on in the issue. IIUC you tried to simplify the example, but it seems to me you may have simplified too much.